validators/types.go

```diff
@@ -0,0 +1,65 @@
+package validators
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+var (
+	ErrMismatchValidatorType    = errors.New("mismatch between validator and validator set")
+	ErrMismatchValidatorSetType = errors.New("mismatch between validator sets")
+	ErrValidatorAlreadyExists   = errors.New("validator already exists in validator set")
+	ErrValidatorNotFound        = errors.New("validator not found in validator set")
+)
+
+type ValidatorType string
+
+const (
+	ECDSAValidatorType ValidatorType = "ecdsa"
+	BLSValidatorType   ValidatorType = "bls"
+)
+
+func (t *ValidatorType) FromString(s string) error {
+	x := ValidatorType(s)
+
+	switch x {
+	case ECDSAValidatorType, BLSValidatorType:
+		*t = x
+
+		return nil
+	default:
+		return fmt.Errorf("invalid validator type: %s", s)
+	}
+}
+
+// Validator defines the interface of the methods a validator implements
+type Validator interface {
+	Type() ValidatorType
+	String() string
+	Addr() types.Address
+	Copy() Validator
+	Equal(Validator) bool
+	MarshalRLPWith(*fastrlp.Arena) *fastrlp.Value
+	UnmarshalRLPFrom(*fastrlp.Parser, *fastrlp.Value) error
+	Bytes() []byte
+	SetFromBytes([]byte) error
+}
+
+// ValidatorSet defines the interface of the methods validator set implements
+type Validators interface {
```

Comment:
I'm really not a big fan of naming this `Validators` and referring to it as the `Validator set`.
Why not just name it `ValidatorSet`?

cc @dbrajovic

Reply:
Then what do you prefer to name? There are 2 types used
(1) A collection of validator
(2) A struct that returns (1) at specified height
@zivkovicmilos

Reply:
Could you give me nice suggestions?

Reply:
@zivkovicmilos Do you need additional discussion?

Reply:
I'm fine with going with this for now